File: seunghwan/programmers/42862/P42862체육복.java
public class Solution {

	public int solution(int n, int[] lost, int[] reserve) {
		int answer = n;
		int clothes[] = new int[n+2];
		for (int i : lost)
			clothes[i]--;
		for (int i : reserve)
			clothes[i]++;
		for (int i = 1; i < n+1; i++) {
			if (clothes[i] == -1)
				if (clothes[i - 1] == 1)
					clothes[i - 1] = clothes[i] = 0;
				else if (clothes[i + 1] == 1)
					clothes[i + 1] = clothes[i] = 0;
				else
					answer--;
		}
		return answer;
	}

}

File: juneyoung/programmers/42862/P42862체육복.kt
class P42862체육복 {
    fun solution(n: Int, lost: IntArray, reserve: IntArray): Int {
        var answer = n
        val reserveSet: HashSet<Int> = hashSetOf()
        val v = Array(n + 1) { true }

        for (i in lost) {
            v[i] = false
            answer--
        }

        for (idx in reserve.indices) {
            reserveSet.add(reserve[idx])
            if (!v[reserve[idx]]) {
                v[reserve[idx]] = true
                reserveSet.remove(reserve[idx])
                answer++
            }
        }

        for (idx in 1..n) {
            if (!v[idx] && reserveSet.contains(idx - 1)) {
                answer++
                reserveSet.remove(idx - 1);
            } else if (!v[idx] && reserveSet.contains(idx + 1)) {
                answer++
                reserveSet.remove(idx + 1);
            }
        }

        return answer
    }
}
 